File: RomanToInt.cpp
#include <iostream>
#include <string>
#include <unordered_map>

using namespace std;
class Solution{
public:
  int romanToInt(string s){
    unordered_map<char, int> roman = {{'I', 1}, {'V', 5}, {'X', 10}, {'L', 50}, {'C', 100}, {'D', 500}, {'M', 1000}};
    int sum = 0;
    int i = 0;
    while(i < s.size()){
      if (roman[s[i]] < roman[s[i+1]])
        sum -= roman[s[i]];
      else
        sum += roman[s[i]];
      i++;
    }
    return sum;
  }
};
int main(){
  Solution sl;
  cout << (sl.romanToInt("MCMXCIV")) << endl;
};

File: RomanToInt.py
class Solution(object):
    def romanToInt(self, s):
        subtraction = {"IV" : 4, "IX" : 9, "XL" : 40, "XC" : 90, "CD" : 400, "CM" : 900}
        sum = 0

        for k,v in subtraction.items():
            if k in s:
                sum += subtraction[k]
                s = s[:s.find(k)] + s[s.find(k)+2:]

        sum += (s.count("I")) + (s.count("V"))*5  + (s.count("X"))*10 + (s.count("L"))*50 + (s.count("C"))*100 + (s.count("D"))*500 + (s.count("M"))*1000
        return sum
